src/library.js

```diff
@@ -4112,7 +4112,11 @@ LibraryManager.library = {
   },
 
   emscripten_run_script_string: function(ptr) {
-    {{{ makeEval("var s = eval(UTF8ToString(ptr)) + '';") }}}
+    {{{ makeEval("var s = eval(UTF8ToString(ptr));") }}}
+    if (s == null) {
+      return null;
+    }
+    s += '';
```

Comment:
The function should return an integer pointer to a C string on the wasm heap, so returning `0` instead of `null` would be better? The return value will be coerced from `null` to `0` anyways.